daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/SpecifiedLengthUnparsers.scala

```diff
@@ -339,6 +339,19 @@ trait KnownPrefixedLengthUnparserMixin {
 
     plElement.setDataValue(java.lang.Integer.valueOf(adjustedLenInUnits.toInt))
 
+    // do checks on facets expressed on prefixLengthType
+    val optSTRD = plElement.erd.optSimpleTypeRuntimeData
+    if (optSTRD.isDefined) {
+      val strd = optSTRD.get
+      val check = strd.executeCheck(plElement)
+      if (check.isError) {
+        val ue = state.toProcessingError(
+          s"The calculated value of ${prefixedLengthERD.namedQName} ($adjustedLenInUnits) failed check due to ${check.errMsg}"
+        )
+        state.setFailed(ue)
```

Comment:
I'm not sure setFailed does the right thing when unparsing? Uparsing doesn't have a concept of backtracking so I think any processing errors should be thrown. I'm not sure setFailed does that. I think most unparsers use the `UE` function?